jobs/etl_jobs/internal/export_clickhouse/core/update.py

```diff
@@ -66,19 +66,52 @@ def remove_stale_partitions(dataset_name, table_name, update_date) -> None:
 def update_overwrite(
     dataset_name: str, table_name: str, tmp_table_name: str, update_date: str
 ) -> None:
+    """
+    Overwrites data in `dataset_name.table_name` either by:
+        - Replacing the specified partition (if the table is partitioned), or
+        - Truncating and reloading the entire table (if it's single-partition).
+    """
     print(f"Will overwrite {dataset_name}.{table_name}. New update : {update_date}")
-    CLICKHOUSE_CLIENT.command(
-        f" ALTER TABLE {dataset_name}.{table_name} ON cluster default REPLACE PARTITION '{update_date}' FROM tmp.{tmp_table_name}"
+
+    # 1) Inspect the table’s partition_key expression
+    partition_key_expr = CLICKHOUSE_CLIENT.command(
+        f"SELECT partition_key FROM system.tables "
+        f"WHERE database='{dataset_name}' AND name='{table_name}'"
+    ).strip()
+    # 'tuple()' or empty means a single dummy partition
+    num_parts = (
+        0
+        if not partition_key_expr or partition_key_expr == "tuple()"
+        else partition_key_expr.count(",") + 1
     )
 
-    remove_stale_partitions(dataset_name, table_name, update_date)
+    if num_parts == 0:
```

Comment:
Detail:
J'aurais tendance a supprimé `num_parts` qui n'est utilisé que pour la condition du if else
```suggestion
    # 'tuple()' or empty means a single dummy partition
    if not partition_key_expr or partition_key_expr == "tuple()":
        ...
    else:
        ...
```